src/api/java/com/minecolonies/api/compatibility/dynamictrees/DynamicTreeCompat.java

```diff
@@ -0,0 +1,277 @@
+      @NotNull final BlockPos pos,
+      @NotNull final IBlockState blockState,
+      @NotNull final int fortune,
+      @NotNull final Block leaf)
+    {
+        if (isDynamicLeavesBlock(leaf))
+        {
+            final NonNullList<ItemStack> list = NonNullList.create();
+            list.addAll(((BlockDynamicLeaves) leaf).getDrops(world, pos, blockState, fortune));
+            return list;
+        }
+        return NonNullList.create();
+    }
+
+    /**
+     * Returns drops of a dynamic seed as List
+     *
+     * @param world      world the Leaf is in
+     * @param pos        position of the Leaf
+     * @param blockState Blockstate of the Leaf
+     * @param fortune    amount of fortune to use
+     * @param leaf       The leaf to check
+     */
+    public static NonNullList<ItemStack> getDropsForLeafCompat(final IBlockAccess world, final BlockPos pos, final IBlockState blockState, final int fortune, final Block leaf)
+    {
+        return instance.getDropsForLeaf(world, pos, blockState, fortune, leaf);
+    }
+
+    /**
+     * Check wether the item is a dynamic Sapling
+     *
+     * @param item Item to check
+     */
+    @Override
+    @Optional.Method(modid = DYNAMIC_MODID)
+    protected boolean checkForDynamicSapling(@NotNull final Item item)
+    {
+        return (item instanceof Seed);
+    }
+
+    /**
+     * Check wether the item is a dynamic Sapling
+     *
+     * @param item Item to check
+     */
+    public static boolean isDynamicTreeSapling(final Item item)
+    {
+        return instance.checkForDynamicSapling(item);
+    }
+
+    /**
+     * Check wether the Itemstack is a dynamic Sapling
+     *
+     * @param stack Itemstack to check
+     * @return true if it is a dynamic Sapling
+     */
+    public static boolean isDynamicTreeSapling(final ItemStack stack)
+    {
+        return instance.checkForDynamicSapling(stack.getItem());
+    }
+
+    /**
+     * Creates a runnable to harvest/break a dynamic tree
+     *
+     * @param world        The world the tree is in
+     * @param blockToBreak The block of the dynamic tree
+     * @param toolToUse    The tool to break the tree with, optional
+     * @param workerPos    The position the fakeplayer breaks the tree from, optional
+     * @return Runnable to break the Tree
+     */
+    @Override
+    @Optional.Method(modid = DYNAMIC_MODID)
+    protected Runnable getTreeBreakActionCompat(@NotNull final World world, @NotNull final BlockPos blockToBreak, final ItemStack toolToUse, final BlockPos workerPos)
+    {
+        return () ->
+        {
+            final IBlockState curBlockState = world.getBlockState(blockToBreak);
+            @Nullable final Block curBlock = curBlockState.getBlock();
+
+            if (world.getMinecraftServer() == null)
+            {
+                Log.getLogger().error("Minecolonies:DynamicTreeCompat unexpected null while trying to get World");
+                return;
+            }
+            final FakePlayer fake =
```

Comment:
Can't we create it static in this class? Maybe even something like "if not null create it, else use it"

Reply:
done 👍